extensions-core/azure-extensions/src/main/java/org/apache/druid/data/input/azure/AzureStorageAccountInputSource.java

```diff
@@ -0,0 +1,253 @@
+        null,
+        split.get(),
+        getObjectGlob(),
+        azureInputSourceConfig,
+        systemFields
+    );
+  }
+
+  @Override
+  public Object getSystemFieldValue(InputEntity entity, SystemField field)
+  {
+    final AzureEntity azureEntity = (AzureEntity) entity;
+
+    switch (field) {
+      case URI:
+        return azureEntity.getUri().toString();
+      case BUCKET:
+        return azureEntity.getLocation().getBucket();
+      case PATH:
+        return azureEntity.getLocation().getPath();
+      default:
+        return null;
+    }
+  }
+
+  @Override
+  protected AzureEntity createEntity(CloudObjectLocation location)
+  {
+    return entityFactory.create(
+        location,
+        new AzureStorage(azureIngestClientFactory, location.getBucket()),
+        SCHEME
+    );
+  }
+
+  @Override
+  protected CloudObjectSplitWidget getSplitWidget()
+  {
+    class SplitWidget implements CloudObjectSplitWidget
+    {
+      @Override
+      public Iterator<LocationWithSize> getDescriptorIteratorForPrefixes(List<URI> prefixes)
+      {
+        return Iterators.transform(
+            azureCloudBlobIterableFactory.create(prefixes, inputDataConfig.getMaxListingLength(), new AzureStorage(azureIngestClientFactory, null)).iterator(),
+            blob -> {
+              try {
+                return new LocationWithSize(
+                    blob.getStorageAccount(),
+                    blob.getContainerName() + "/" + blob.getName(),
+                    blob.getBlobLength()
+                );
+              }
+              catch (BlobStorageException e) {
+                throw new RuntimeException(e);
+              }
+            }
+        );
+      }
+
+      @Override
+      public long getObjectSize(CloudObjectLocation location)
+      {
+        try {
+          AzureStorage azureStorage = new AzureStorage(azureIngestClientFactory, location.getBucket());
+          Pair<String, String> locationInfo = getContainerAndPathFromObjectLocation(location);
+          final BlockBlobClient blobWithAttributes = azureStorage.getBlockBlobReferenceWithAttributes(
+              locationInfo.lhs,
+              locationInfo.rhs
+          );
+
+          return blobWithAttributes.getProperties().getBlobSize();
+        }
+        catch (BlobStorageException e) {
+          throw new RuntimeException(e);
+        }
+      }
+    }
+
+    return new SplitWidget();
+  }
+
+  @Nullable
+  @JsonProperty("properties")
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  public AzureInputSourceConfig getAzureInputSourceConfig()
+  {
+    return azureInputSourceConfig;
+  }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    if (!super.equals(o)) {
+      return false;
+    }
+    AzureStorageAccountInputSource that = (AzureStorageAccountInputSource) o;
+    return entityFactory.equals(that.entityFactory) &&
```

Comment:
Should the factory methods be compared for equals? Also what about the prefixes and other fields specified in the `toString()` method below, do they belong here as well?

Reply:
the equalsVerifier gets mad if we don't include them

Reply:
Could ignore them with something like this https://github.com/apache/druid/blob/06b228ff7cbdd7d9bc7dad16e7c988bedaae1009/processing/src/test/java/org/apache/druid/segment/filter/ArrayContainsElementFilterTests.java#L750